support-frontend/assets/pages/contributions-landing/components/ContributionForm.tsx

```diff
@@ -230,158 +227,155 @@ const mapDispatchToProps = (dispatch: (...args: any[]) => any) => ({
+					accountHolderName,
+					iban,
+					country,
+					streetName,
 				});
-			} else {
-				logException('Missing orderReferenceId for amazon pay');
 			}
 		},
-		None: () => {
-			logInvalidCombination('ONE_OFF', 'None');
+		ExistingCard: () =>
+			props.onPaymentAuthorisation({
+				paymentMethod: 'ExistingCard',
+				billingAccountId: props.existingPaymentMethod?.billingAccountId ?? '',
+			}),
+		ExistingDirectDebit: () =>
+			props.onPaymentAuthorisation({
+				paymentMethod: 'ExistingDirectDebit',
+				billingAccountId: props.existingPaymentMethod?.billingAccountId ?? '',
+			}),
+		AmazonPay: () => {
+			if (props.amazonPayBillingAgreementId) {
+				props.onPaymentAuthorisation({
+					paymentMethod: 'AmazonPay',
+					amazonPayBillingAgreementId: props.amazonPayBillingAgreementId,
+				});
+			}
 		},
-	},
-	ANNUAL: {
-		...formHandlersForRecurring,
-		None: () => {
-			logInvalidCombination('ANNUAL', 'None');
+	};
+	const formHandlers: PaymentMatrix<() => void> = {
+		ONE_OFF: {
+			Stripe: () => {
+				if (createStripePaymentMethod) {
+					createStripePaymentMethod(null);
+				}
+			},
+			PayPal: () => {
+				props.setPaymentIsWaiting(true);
+				props.createOneOffPayPalPayment({
+					currency: props.currency,
+					amount: getAmount(
+						props.selectedAmounts,
+						props.otherAmounts,
+						props.contributionType,
+					),
+					returnURL: payPalReturnUrl(props.countryGroupId, props.email),
+					cancelURL: payPalCancelUrl(props.countryGroupId),
+				});
+			},
+			DirectDebit: () => {
+				logInvalidCombination('ONE_OFF', DirectDebit);
+			},
+			Sepa: () => {
+				logInvalidCombination('ONE_OFF', Sepa);
+			},
+			ExistingCard: () => {
+				logInvalidCombination('ONE_OFF', ExistingCard);
+			},
+			ExistingDirectDebit: () => {
+				logInvalidCombination('ONE_OFF', ExistingDirectDebit);
+			},
+			AmazonPay: () => {
+				const { amazonPayOrderReferenceId } = props;
+
+				if (amazonPayOrderReferenceId) {
+					props.setPaymentIsWaiting(true);
+					props.onPaymentAuthorisation({
+						paymentMethod: AmazonPay,
+						orderReferenceId: amazonPayOrderReferenceId,
+					});
+				} else {
+					logException('Missing orderReferenceId for amazon pay');
+				}
+			},
+			None: () => {
+				logInvalidCombination('ONE_OFF', 'None');
+			},
 		},
-	},
-	MONTHLY: {
-		...formHandlersForRecurring,
-		None: () => {
-			logInvalidCombination('MONTHLY', 'None');
+		ANNUAL: {
+			...formHandlersForRecurring,
+			None: () => {
+				logInvalidCombination('ANNUAL', 'None');
+			},
 		},
-	},
-};
+		MONTHLY: {
+			...formHandlersForRecurring,
+			None: () => {
+				logInvalidCombination('MONTHLY', 'None');
+			},
+		},
+	};
 
-// Note PayPal recurring flow does not call this function
-function onSubmit(
-	props: PropTypes,
-): (event: React.FormEvent<HTMLFormElement>) => void {
-	return (event) => {
+	// Note PayPal recurring flow does not call this function
+	function onSubmit(event: React.FormEvent<HTMLFormElement>): void {
```

Comment:
This also got moved inside the component as it uses the `formHandlers` object. This means it can also drop the need to take `props` as an argument